packages/core/admin/admin/src/components/Filters.tsx

```diff
@@ -0,0 +1,573 @@
+ * Root
+ * -----------------------------------------------------------------------------------------------*/
+
+interface FilterFormData {
+  name: string;
+  filter: string;
+  value?: string;
+}
+
+interface FitlersContextValue {
+  disabled: boolean;
+  onChange: (data: FilterFormData) => void;
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  setTriggerNode: (node: HTMLButtonElement | null) => void;
+  triggerNode: HTMLButtonElement | null;
+  options: Filters.Filter[];
+}
+
+const [FiltersProvider, useFilters] = createContext<FitlersContextValue>('Filters');
+
+interface RootProps
+  extends Partial<Pick<FitlersContextValue, 'disabled' | 'onChange' | 'options' | 'open'>> {
+  children: React.ReactNode;
+  defaultOpen?: boolean;
+  onOpenChange?: (open: boolean) => void;
+}
+
+const Root = ({
+  children,
+  disabled = false,
+  onChange,
+  onOpenChange,
+  open: openProp,
+  defaultOpen,
+  options = [],
+}: RootProps) => {
+  const [triggerNode, setTriggerNode] = React.useState<FitlersContextValue['triggerNode']>(null);
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  const handleChange = (data: FilterFormData) => {
+    if (onChange) {
+      onChange(data);
+    }
+  };
+
+  return (
+    <FiltersProvider
+      disabled={disabled}
+      onChange={handleChange}
+      open={open}
+      options={options}
+      setOpen={setOpen}
+      setTriggerNode={setTriggerNode}
+      triggerNode={triggerNode}
+    >
+      {children}
+    </FiltersProvider>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * Trigger
+ * -----------------------------------------------------------------------------------------------*/
+
+const Trigger = React.forwardRef<HTMLButtonElement, Filters.TriggerProps>(
+  ({ label }, forwardedRef) => {
+    const { formatMessage } = useIntl();
+    const { setTriggerNode, setOpen } = useFilters('Trigger', ({ setTriggerNode, setOpen }) => ({
+      setTriggerNode,
+      setOpen,
+    }));
+    const disabled = useFilters('Trigger', ({ disabled }) => disabled);
+    const open = useFilters('Trigger', ({ open }) => open);
+
+    const composedRefs = useComposedRefs(forwardedRef, setTriggerNode);
+
+    const handleClick = () => setOpen(!open);
+
+    return (
+      <Button
+        variant="tertiary"
+        ref={composedRefs}
+        startIcon={<FilterIcon />}
+        onClick={handleClick}
+        size="S"
+        disabled={disabled}
+      >
+        {label || formatMessage({ id: 'app.utils.filters', defaultMessage: 'Filters' })}
+      </Button>
+    );
+  }
+);
+
+/* -------------------------------------------------------------------------------------------------
+ * Popover
+ * -----------------------------------------------------------------------------------------------*/
+
+const PopoverImpl = () => {
```

Comment:
This name confused me a little bit, it is not better something like "FilterPopover"?

Reply:
Isn't that a bit redundant considering we're already in the `Filters` file? It would be used as `Filters.Popover` in the end, It's called PopoverImpl because it's an implementation of the Popover, but we have one in the DS lol, I could swap those around though so this just becomes Popover?

Reply:
ok I think we can stay with PopoverImpl